test/e2e/tests/forms.spec.js

```diff
@@ -187,8 +189,123 @@ for (let path of ["/form/nested", "/form"]) {
+  let testForm = page.locator("#test-form");
+  let testInput = page.locator("#test-form input[name=myname]");
+  let submitBtn = page.locator("#test-form button");
+  // initial 3 stream items
+  await expect(page.locator("#form-stream li")).toHaveCount(3)
+  await testInput.fill("1");
+  await testInput.fill("2");
+  // form is locked on phx-change and stream remains unchanged
+  await sleep(1000)
+  await expect(testForm).toHaveClass("phx-change-loading");
+  await expect(testInput).toHaveClass("phx-change-loading");
+  await expect(testForm).toHaveAttribute("data-phx-ref-loading");
+  await expect(testForm).toHaveAttribute("data-phx-ref-src");
+  await expect(testInput).toHaveAttribute("data-phx-ref-loading");
+  await expect(testInput).toHaveAttribute("data-phx-ref-src");
+  await expect(page.locator("#form-stream li")).toHaveCount(3, {timeout: 100});
+  // on unlock, cloned stream items that are added on each phx-change are applied to DOM
+  await expect(page.locator("#form-stream li")).toHaveCount(5);
+  // after clones are applied, the stream item hooks are mounted
+  await expect(page.locator("#form-stream li")).toHaveText([
+    "*%{id: 1}pong",
+    "*%{id: 2}pong",
+    "*%{id: 3}pong",
+    "*%{id: 4}pong",
+    "*%{id: 5}pong"
+  ]);
+  await submitBtn.click();
+  await expect(submitBtn).toHaveText("Saving...");
+  await expect(testForm).toHaveClass("phx-submit-loading");
+  await expect(testInput).toHaveAttribute("readonly", "");
+  await expect(submitBtn).toHaveClass("phx-submit-loading");
+  await expect(testForm).toHaveAttribute("data-phx-ref-loading");
+  await expect(testForm).toHaveAttribute("data-phx-ref-src");
+  await expect(testInput).toHaveAttribute("data-phx-ref-loading");
+  await expect(testInput).toHaveAttribute("data-phx-ref-src");
+  await expect(submitBtn).toHaveAttribute("data-phx-ref-loading");
+  await expect(submitBtn).toHaveAttribute("data-phx-ref-src");
+  // submit adds 1 more stream item and new hook is mounted
+  await expect(page.locator("#form-stream li")).toHaveText([
+    "*%{id: 1}pong",
+    "*%{id: 2}pong",
+    "*%{id: 3}pong",
+    "*%{id: 4}pong",
+    "*%{id: 5}pong",
+    "*%{id: 6}pong"
+  ]);
+  await expect(submitBtn).toHaveText("Submit");
+  await expect(submitBtn).toHaveAttribute("phx-disable-with", "Saving...");
+  await expect(testForm).not.toHaveClass("phx-submit-loading");
+  await expect(testInput).not.toHaveAttribute("readonly");
+  await expect(submitBtn).not.toHaveClass("phx-submit-loading");
+  await expect(testForm).not.toHaveAttribute("data-phx-ref-loading");
+  await expect(testForm).not.toHaveAttribute("data-phx-ref-src");
+  await expect(testInput).not.toHaveAttribute("data-phx-ref-loading");
+  await expect(testInput).not.toHaveAttribute("data-phx-ref-src");
+  await expect(submitBtn).not.toHaveAttribute("data-phx-ref-loading");
+  await expect(submitBtn).not.toHaveAttribute("data-phx-ref-src");
+  await page.evaluate(() => window.liveSocket.disableLatencySim());
+});
```

Comment:
If you change `data-phx-ref-loading` to `data-phx-ref`, this test already works on main. I think it would make sense to change the order here to click on submit before the phx-change ack arrives. Then the test could check that the stream gets the 5 pongs while the form is already in the submit state.

```suggestion
test(`loading and locked states with latent clone`, async ({ page }) => {
  await page.goto(`/form/stream`);
  let formHook = page.locator("#form-stream-hook");
  await syncLV(page);
  await page.evaluate(() => window.liveSocket.enableLatencySim(2000));
  await expect(formHook).toHaveText("pong");
  let testForm = page.locator("#test-form");
  let testInput = page.locator("#test-form input[name=myname]");
  let submitBtn = page.locator("#test-form button");
  // initial 3 stream items
  await expect(page.locator("#form-stream li")).toHaveCount(3)
  await testInput.fill("1");
  await testInput.fill("2");
  // form is locked on phx-change and stream remains unchanged
  await sleep(1000)
  await submitBtn.click();
  await expect(testForm).toHaveClass("phx-change-loading phx-submit-loading");
  await expect(submitBtn).toHaveText("Saving...");
  await expect(testInput).toHaveClass("phx-change-loading");
  await expect(testForm).toHaveAttribute("data-phx-ref-loading");
  await expect(testForm).toHaveAttribute("data-phx-ref-src");
  await expect(testInput).toHaveAttribute("data-phx-ref-loading");
  await expect(testInput).toHaveAttribute("data-phx-ref-src");
  await expect(page.locator("#form-stream li")).toHaveCount(3, {timeout: 100});
  // on unlock, cloned stream items that are added on each phx-change are applied to DOM
  await expect(page.locator("#form-stream li")).toHaveCount(5);
  // after clones are applied, the stream item hooks are mounted
  // note that the form still awaits the submit ack, but it is not locked,
  // therefore the updates from the phx-change are already applied
  await expect(page.locator("#form-stream li")).toHaveText([
    "*%{id: 1}pong",
    "*%{id: 2}pong",
    "*%{id: 3}pong",
    "*%{id: 4}",
    "*%{id: 5}"
  ]);
  // still saving
  await expect(submitBtn).toHaveText("Saving...");
  await expect(testForm).toHaveClass("phx-submit-loading");
  await expect(testInput).toHaveAttribute("readonly", "");
  await expect(submitBtn).toHaveClass("phx-submit-loading");
  await expect(testForm).toHaveAttribute("data-phx-ref-loading");
  await expect(testForm).toHaveAttribute("data-phx-ref-src");
  await expect(testInput).toHaveAttribute("data-phx-ref-loading");
  await expect(testInput).toHaveAttribute("data-phx-ref-src");
  await expect(submitBtn).toHaveAttribute("data-phx-ref-loading");
  await expect(submitBtn).toHaveAttribute("data-phx-ref-src");
  // submit adds 1 more stream item and new hook is mounted
  await expect(page.locator("#form-stream li")).toHaveText([
    "*%{id: 1}pong",
    "*%{id: 2}pong",
    "*%{id: 3}pong",
    "*%{id: 4}pong",
    "*%{id: 5}pong",
    "*%{id: 6}pong"
  ]);
  await expect(submitBtn).toHaveText("Submit");
  await expect(submitBtn).toHaveAttribute("phx-disable-with", "Saving...");
  await expect(testForm).not.toHaveClass("phx-submit-loading");
  await expect(testInput).not.toHaveAttribute("readonly");
  await expect(submitBtn).not.toHaveClass("phx-submit-loading");
  await expect(testForm).not.toHaveAttribute("data-phx-ref");
  await expect(testForm).not.toHaveAttribute("data-phx-ref-src");
  await expect(testInput).not.toHaveAttribute("data-phx-ref");
  await expect(testInput).not.toHaveAttribute("data-phx-ref-src");
  await expect(submitBtn).not.toHaveAttribute("data-phx-ref");
  await expect(submitBtn).not.toHaveAttribute("data-phx-ref-src");
  await page.evaluate(() => window.liveSocket.disableLatencySim());
});
```